components/hierarchical-view/hierarchical-view-mixin.js

```diff
@@ -0,0 +1,380 @@
+			return;
+		}
+		let rect;
+		if (view === this) {
+			rect = this.shadowRoot.querySelector('.d2l-hierarchical-view-content').getBoundingClientRect();
+		} else {
+			rect = view.getBoundingClientRect();
+		}
+		this.style.height = `${rect.height}px`;
+
+	}
+
+	__fireShowComplete(data) {
+		const eventDetails = {
+			bubbles: true,
+			composed: true,
+			detail: { activeView: this.getActiveView(), data: data }
+		};
+		this.dispatchEvent(new CustomEvent('d2l-hierarchical-view-show-complete', eventDetails));
+	}
+
+	__fireHideComplete(data) {
+		const eventDetails = {
+			bubbles: true,
+			composed: true,
+			detail: { activeView: this.getActiveView(), data: data }
+		};
+		this.dispatchEvent(new CustomEvent('d2l-hierarchical-view-hide-complete', eventDetails));
+	}
+
+	__fireViewResize() {
+		if (!this.isActive()) {
+			const view = this.getActiveView();
+			view.resize();
+			return;
+		}
+		this.__content = this.shadowRoot.querySelector('.d2l-hierarchical-view-content');
+		this.__bound_reactToChanges = this.__bound_reactToChanges || this.__reactToChanges.bind(this);
+		this.__resizeObserver = this.__resizeObserver || new ResizeObserver(this.__bound_reactToChanges);
+		this.__resizeObserver.disconnect();
+		this.__resizeObserver.observe(this.__content);
+	}
+
+	__reactToChanges() {
+		const contentRect = this.__content.getBoundingClientRect();
+		if (contentRect.height < 1) return;
+		const eventDetails = {
+			bubbles: true,
+			composed: true,
+			detail: contentRect
+		};
+		this.dispatchEvent(new CustomEvent('d2l-hierarchical-view-resize', eventDetails));
+	}
+
+	__focusCapture(e) {
+		const parentView = this.__getParentViewFromEvent(e);
+
+		if (parentView.isActive()) {
+			return;
+		}
+
+		const relatedTarget = e.relatedTarget;
+		let focusableElement;
+
+		const getNextFocusableLocal = () => {
+			const activeView = this.getActiveView();
+			if (this.__nativeFocus === activeView.focus) {
+				return getNextFocusable(activeView);
+			} else {
+				return activeView;
+			}
+		};
+
+		if (relatedTarget) {
+			if (!isComposedAncestor(this, relatedTarget)) {
+				focusableElement = getNextFocusableLocal();
+			} else {
+				focusableElement = getPreviousFocusable(this);
+			}
+		} else {
+
+			// handle focus for ie
+			if (this.__focusPrevious) {
+				this.__focusPrevious = false;
+				focusableElement = getPreviousFocusable(this);
+			} else {
+				focusableElement = getNextFocusableLocal();
+			}
+
+		}
+
+		if (focusableElement) {
+			focusableElement.focus();
+		}
+
+	}
+
+	__focusOutCapture(e) {
+		// focus tracking required since ie only supports relatedTarget on focusin/focusout
+		const relatedTarget = e.relatedTarget;
+		const activeView = this.getActiveView();
+		this.__focusPrevious = false;
+		if (isComposedAncestor(activeView, e.target)) {
+			if (isComposedAncestor(this, relatedTarget)) {
+				this.__focusPrevious = true;
+			}
+		}
+
+	}
+
+	__getParentViewFromEvent(e) {
+		const path = e.path;
```

Comment:
I am suspicious of this `path` property.  Is this working as expected?  The reason I ask is that the original code was doing `dom(e).path`, and so Polymer was providing that property on a normalized event object.  I think the standard cross browser API is `composedPath()`, which similarly returns an array... the first element being the root element of the event. Or is there a standard `path` as well... that is different in some way from `composedPath`?

Reply:
It does seem to be functioning the same as before. I'll do some cross-browser testing though to make sure all is good.